File: chapitre-05/exemple-assert.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-assert.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function

def succeeding_function(num):
	assert (num >= 0), "Le nombre doit etre positif ou nul"
	print("Ok,", num, "est positif")

def failing_function(num):
	assert (num <= 0), "Le nombre doit etre negatif ou nul"
	print("Ok,", num, "est negatif")



succeeding_function(5)
failing_function(5)

File: chapitre-05/exemple-assert.c
// ------------------------------------------------------------------
// exemple-assert.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>


void succeeding_function(int i);
void failing_function(int i);

int main(void)
{
	succeeding_function(5);
	failing_function(5);

	return EXIT_SUCCESS;
}

void succeeding_function(int num)
{
	/* Cette fonction necessite que i soit positif */
	assert(num >= 0);
	fprintf(stdout, "Ok, num est positif\n");
}

void failing_function(int num)
{
	/* Cette fonction necessite que i soit negatif */
	assert(num <= 0);
	fprintf(stdout, "Ok, num est negatif\n");
}

